### ISSUE TITLE
Is this outdated?

### ISSUE THREAD

**Issue by tupiasplyths** [jonegil/gui-with-gio#36]
Is this tutorial still up to date? I can't run the source code because several variables got marked as "declared and not used" and 
```
select {
// listen for events in the window.
case e := <-w.Events():
              ^^^^^
	// detect what type of event
	switch e := e.(type) {
```
is undefined.
Sorry if this question is silly, I'm very new to Go and Gio UI.

**Comment by jonegil** [comment]
Let me take it for a new spin. 

Thanks for reporting.

**Comment by zilllaiss** [comment]
yeah, the tutorial is outdated. I still manage to adapt some code by reading the doc, but I can't manage for all of them to work, especially I'm stuck at https://jonegil.github.io/gui-with-gio/egg_timer/07_progressbar.html since I couldn't manage to make the button and progress bar reactive no matter what I do :/

it would be nice to have an update soon

**Comment by jonegil** [comment]
7 chapters done, 4 to go

**Comment by zilllaiss** [comment]
wow, thank you for your hardwork! it helped me so much. I realized that I only messed up some order of writing after all this time,.. silly me. I finished building the app despite it being a little bit out of date, thanks to your intuitive tutorial.

although, it seems to left out some code to put on the site at chapter 10, as it only appears on github. 
```go
// resetting the boil
if progress >= 1 {
    progress = 0
}

// end precisely on 1
if progress >= 1 {
    progress = 1
}
```

but not a big deal as I found in the end.
